snuba/datasets/tags_column_processor.py

```diff
@@ -77,38 +125,135 @@ def __string_col(self, col: str) -> str:
         else:
             return "toString({})".format(escape_identifier(col))
 
-    def __tag_expr(self, column_name: str, table_alias: str = "",) -> str:
+    def __tag_expr(self, parsed_col: ParsedNestedColumn, table_alias: str = "",) -> str:
         """
         Return an expression for the value of a single named tag.
 
         For tags/contexts, we expand the expression depending on whether the tag is
         "promoted" to a top level column, or whether we have to look in the tags map.
         """
-        col, tag = NESTED_COL_EXPR_RE.match(column_name).group(1, 2)
         # For promoted tags, return the column name.
+        assert parsed_col.parameter
```

Comment:
It seems like in an ideal situation, this method would a separate type where this attribute was not optional.

Reply:
We will not need this because tags/contexts processing will be split from tags_key/tags_value, so that whole class will not exist